source/server-discovery-and-monitoring/server-discovery-and-monitoring-tests.md

```diff
@@ -172,3 +172,39 @@ This test requires failCommand appName support which is only available in MongoD
 5. Then verify that a ServerHeartbeatSucceededEvent and a ConnectionPoolReadyEvent (CMAP) are emitted.
 
 6. Disable the failpoint.
+
+## Connection Pool Backpressure
+
+This test will be used to ensure that connection establishment failures during the TLS handshake do not result in a pool
+clear event. We create a setup client to enable the ingress connection establishment rate limiter, and then induce a
+connection storm. After the storm, we verify that some of the connections failed to checkout, but that the pool was not
+cleared.
+
+This test requires MongoDB 7.0+.
+
+1. Create a setup client and run the following commands to set up the rate limiter:
+
+    ```python
+    db = admin_client.admin
```

Comment:
question: does setParameter replicate across shards? or must we set the parameter on every mongos in a sharded cluster, like failCommands?

Reply:
Yeah, let's do that: "The number of new ingress connections that will be allowed to establish per second"